internal/config/config.go

```diff
@@ -169,7 +175,7 @@ func validateHoldTime(ht time.Duration) error {
 }
 
 // Parse loads and validates a Config from bs.
-func For(resources ClusterResources, validate Validate) (*Config, error) {
+func For(ctx context.Context, cli client.Client, resources ClusterResources, validate Validate) (*Config, error) {
```

Comment:
This function is pure, meaning it has no side effects and the result does not depend on the system state.
It's important that stays this way, because otherwise we'd need to inject a mock of the client to test it.

So, we need to collect the secrets upfront and pass them here, in a way or another.

Reply:
sure, done.